src/main/java/org/wildfly/security/hashicorp/vault/VaultConnector.java

```diff
@@ -28,13 +28,23 @@ public class VaultConnector {
     private final SslConfig sslConfig;
     private Vault vault;
 
+    private JwtConfig jwtConfig;
+
     public VaultConnector(String vaultUrl, String token, String namespace, SslConfig sslConfig, boolean sslVerify) {
         this.vaultUrl = vaultUrl;
         this.token = token;
         this.namespace = namespace;
         this.sslConfig = sslConfig;
     }
 
+    public VaultConnector(String vaultUrl, JwtConfig jwtConfig, String namespace, SslConfig sslConfig) {
+        this.vaultUrl = vaultUrl;
+        this.token = "";
```

Comment:
@honza-kasik Can't we have both token and jwt configured? In either case, I wouldn't force empty string here, it seems better to let it be null so it is clear that it was not configured to be empty by user